relay/kusama/src/impls.rs

```diff
@@ -0,0 +1,177 @@
+		let para_byte_deposit = deposit(0, 1) / 100;
+		let para_sub_account_deposit = deposit(1, 53) / 100;
+		let para_existential_deposit = EXISTENTIAL_DEPOSIT / 10;
+
+		// pallet deposits
+		let id_deposit =
+			para_basic_deposit.saturating_add(para_byte_deposit.saturating_mul(bytes as Balance));
+		let subs_deposit = para_sub_account_deposit.saturating_mul(subs as Balance);
+
+		id_deposit
+			.saturating_add(subs_deposit)
+			.saturating_add(para_existential_deposit.saturating_mul(2))
+	}
+}
+
+// Note / Warning: This implementation should only be used in a transactional context. If not, then
+// an error could result in assets being burned.
+impl<Runtime, AccountId> OnReapIdentity<AccountId> for ToParachainIdentityReaper<Runtime, AccountId>
+where
+	Runtime: frame_system::Config + pallet_xcm::Config,
+	AccountId: Into<[u8; 32]> + Clone + Encode,
+{
+	fn on_reap_identity(who: &AccountId, fields: u32, subs: u32) -> DispatchResult {
+		use crate::{
+			impls::IdentityMigratorCalls::PokeDeposit,
+			weights::runtime_common_identity_migrator::WeightInfo as MigratorWeights,
+		};
+
+		let total_to_send = Self::calculate_remote_deposit(fields, subs);
+
+		// define asset / destination from relay perspective
+		let ksm = Asset { id: AssetId(Here.into_location()), fun: Fungible(total_to_send) };
+		// People Chain: ParaId 1004
+		let destination: Location = Location::new(0, Parachain(1004));
+
+		// Do `check_out` accounting since the XCM Executor's `InitiateTeleport` doesn't support
+		// unpaid teleports.
+
+		// withdraw the asset from `who`
+		let who_origin =
+			Junction::AccountId32 { network: None, id: who.clone().into() }.into_location();
+		let _withdrawn = xcm_config::LocalAssetTransactor::withdraw_asset(&ksm, &who_origin, None)
+			.map_err(|err| {
+				defensive!(
+					"runtime::on_reap_identity: withdraw_asset(what: {:?}, who_origin: {:?}) error: {:?}",
+					(&ksm, &who_origin, err)
+				);
+				pallet_xcm::Error::<Runtime>::LowBalance
+			})?;
+
+		// check out
+		xcm_config::LocalAssetTransactor::can_check_out(
+			&destination,
+			&ksm,
+			// not used in AssetTransactor
+			&XcmContext { origin: None, message_id: [0; 32], topic: None },
+		)
+		.map_err(|err| {
+			log::error!(
+				target: "runtime::on_reap_identity",
+				"can_check_out(destination: {:?}, asset: {:?}, _) error: {:?}",
+				destination, ksm, err
+			);
+			pallet_xcm::Error::<Runtime>::CannotCheckOutTeleport
+		})?;
+		xcm_config::LocalAssetTransactor::check_out(
+			&destination,
+			&ksm,
+			// not used in AssetTransactor
+			&XcmContext { origin: None, message_id: [0; 32], topic: None },
+		);
+
+		// reanchor
+		let ksm_reanchored: Assets =
+			vec![Asset { id: AssetId(Location::new(1, Here)), fun: Fungible(total_to_send) }]
+				.into();
+
+		let poke = PeopleRuntimePallets::<AccountId>::IdentityMigrator(PokeDeposit(who.clone()));
+		let remote_weight_limit = MigratorWeights::<Runtime>::poke_deposit().saturating_mul(2);
```

Comment:
why is this multiplied by 2?

Reply:
It is essentially a conservative factor of safety. This Identity Migrator pallet is in both runtimes, but each chain only has access to its own weight info. So this is, "this is _my_ weight for `poke_deposit`, and I assume the remote weight will be less than half". I'm not super concerned with efficiency here as this will only be used when the parachain is in a semi-locked state and users can't actually use it left, so essentially all the block weight is for the system itself to handle setup.